src/main.c

```diff
@@ -875,17 +744,77 @@ static int send_event(enum zl_event_t event_type, void *event_data) {
   return 0;
 }
 
+static int get_component_list(char *buf, size_t buf_size) {
+  char command[4*PATH_MAX];
+  snprintf (command, sizeof(command), "%s/bin/internal/get-launch-components.sh -c %s -r %s",
+    zl_context.root_dir, zl_context.instance_dir, zl_context.root_dir);
+  INFO("about to run command %s\n", command);
+  FILE *fp = popen(command, "r");
+  if (!fp) {
+    ERROR("unable to get start components - %s\n", strerror(errno));
+    return -1;
+  }
+  char *line = fgets(buf, buf_size - 1, fp);
+  fclose(fp);
+  if (!line) {
+    ERROR("failed to read start component list - %s\n", strerror(errno));
+    return -1;
+  }
+  int len = strlen(line);
+  for (int i = len - 1; i >= 0; i--) {
+    if (line[i] != ' ' && line[i] != '\n' && line[i] != ',') {
+      break;
+    }
+    line[i] = '\0';
+  }
+  if (strlen(line) == 0) {
+    ERROR("start component list is empty\n");
+    return -1;
+  }
+  INFO("start component list: '%s'\n", line);
+  return 0;
+}
+
+static int prepare_workspace() {
+  char command[PATH_MAX];
+  INFO("about to prepare zowe workspace\n");
+  const char *script = "bin/internal/prepare-workspace-for-launcher.sh";
```

Comment:
This ties the launcher to the zowe-install-packaging repo. Can we externalize this?

Reply:
Not sure. How could we pass the script name to the launcher? via environment variable?